app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenControllerTest.kt

```diff
@@ -290,29 +254,24 @@ class PatientEntryScreenControllerTest {
   @Test
   fun `validation errors should be cleared on every input change`() {
     uiEvents.onNext(PatientFullNameTextChanged("Ashok"))
-    uiEvents.onNext(PatientNoPhoneNumberToggled(noneSelected = false))
     uiEvents.onNext(PatientPhoneNumberTextChanged("1234567890"))
     uiEvents.onNext(PatientDateOfBirthTextChanged("12/04/1993"))
     uiEvents.onNext(PatientGenderChanged(Just(Gender.TRANSGENDER)))
-    uiEvents.onNext(PatientNoColonyOrVillageToggled(noneSelected = false))
     uiEvents.onNext(PatientColonyOrVillageTextChanged("colony"))
     uiEvents.onNext(PatientDistrictTextChanged("district"))
     uiEvents.onNext(PatientStateTextChanged("state"))
 
     uiEvents.onNext(PatientPhoneNumberTextChanged(""))
-    uiEvents.onNext(PatientNoPhoneNumberToggled(noneSelected = true))
     uiEvents.onNext(PatientDateOfBirthTextChanged(""))
     uiEvents.onNext(PatientAgeTextChanged("20"))
-    uiEvents.onNext(PatientNoColonyOrVillageToggled(noneSelected = true))
     uiEvents.onNext(PatientColonyOrVillageTextChanged(""))
 
     verify(screen).showEmptyFullNameError(false)
-    verify(screen, times(4)).showEmptyPhoneNumberError(false)
     verify(screen, times(3)).showEmptyDateOfBirthAndAgeError(false)
     verify(screen, times(2)).showInvalidDateOfBirthError(false)
     verify(screen, times(2)).showDateOfBirthIsInFutureError(false)
     verify(screen).showMissingGenderError(false)
-    verify(screen, times(4)).showEmptyColonyOrVillageError(false)
+    verify(screen, atLeastOnce()).showEmptyColonyOrVillageError(false)
```

Comment:
When writing this test, I intentionally used specific `times` to ensure that I do not miss any validations. Do you think that's not required?

Reply:
That's the wrong way to verify validations since it creates a tight coupling between the test and the implementation. If we are verifying this, we should be sending a list of errors to the screen and validate that the errors sent to screen are matching what we expect.

Reply:
Hmmmmmmmmmmmm. In that case, do you want to remove `times` from other checks in this function? Otherwise we'll have a mix of both.

Reply:
I'd rather leave it for now, and change the screen api itself in a future PR to handle this change and update the test then. WDYT?

Reply:
That sounds like the ideal case, but let's not leave this function hanging in the middle. I've seen Sanchita use this function as a reference for one of her tests, and it'll only confuse other people more.